src/server.js

```diff
@@ -0,0 +1,59 @@
+require('require-self-ref')
+const {checkWatcher, startNPMWatcher} = require('./watchNPM')
+const redis = require('redis')
+const logger = require('~/src/logging').logger(module)
+const config = require('~/src/config')
+const queue = require('windbreaker-service-util/queue')
+const Promise = require('bluebird')
+config.load()
+
+;(async () => {
+  let connection = null
+  let channel = null
+  let client = null
+  const AmqUrl = config.getAmqUrl()
+
+  const setup = async function () {
+    try {
+      connection = await queue.createConnection({
+        logger,
+        AmqUrl
+      })
+      channel = await connection.createChannel()
+      await channel.assertQueue(config.getQueueName())
+      client = redis.createClient(config.getRedisURL())
+      logger.info('setup successful')
+    } catch (error) {
+      logger.error(error)
+      logger.error('cleaning up and restarting setup')
+      client = null
+      if (channel) channel.close()
+      channel = null
+      if (connection) connection.close()
+      connection = null
+      await Promise.delay(1000).then(async function () {
+        await setup()
+      })
+    }
+  }
```

Comment:
What's going on here? We have this work handled in the `windbreaker-service-util` project. You can use the utils exposed to create a producer for you. It handles all of the message encoding and ensure that the queue definition stays the same between our consumers and producers so that things won't break.